src/Confluent.Kafka/Consumer_KV.cs

```diff
@@ -0,0 +1,385 @@
+// Copyright 2016-2017 Confluent Inc., 2015-2016 Andreas Heider
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Derived from: rdkafka-dotnet, licensed under the 2-clause BSD License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Confluent.Kafka.Impl;
+using Confluent.Kafka.Internal;
+using Confluent.Kafka.Serialization;
+
+
+namespace Confluent.Kafka
+{
+    /// <summary>
```

Comment:
this class was moved here out of `Consumer.cs`. It also includes the following changes:

modifications are:

1. comments have been moved out to include_docs_consumer.xml, include_docs_client.xml
2. I added the `Commit` methods (see note in Consumer.cs).